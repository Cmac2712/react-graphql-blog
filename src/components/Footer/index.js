import React, { Component } from 'react';
import { Wrapper } from '../App/Theme';
import StyledFooter from './style';
import ReactSVG from 'react-svg'

class Footer extends Component {

	getCurrentYear() {
		return new Date().getFullYear();
	}

	render() {
		return(
			<StyledFooter>
				<Wrapper>
					<div className="footer-container">
						<a
							href="https://github.com/Cmac2712/react-graphql-blog"
							className="github"
						>
							<span>View on GitHub</span>
							<ReactSVG
								src="/images/github.svg"
								wrapper="span"
								className="github-icon-container"
								onInjected={(error) => {
									error && console.log(error);
									return;
								}}
								svgClassName="github-icon"
								svgStyle={{
									height: 16, 
									width: 16,
										fill: 'red'
								}}
							/>
						</a>
						<a href="https://craigmacintyre.co.uk/" className="myname">
							Craig MacIntyre
						</a>
						<div className="year">
							 { this.getCurrentYear() }
						</div>
					</div>
				</Wrapper>
			</StyledFooter>
		);
	} 
}

export default Footer;
