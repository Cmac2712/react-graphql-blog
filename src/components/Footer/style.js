import styled from 'styled-components';

const StyledFooter = styled.footer`
	padding: ${props => props.theme.spacingUnit}px;
	background-color: #f1f1f1;
	margin-top: ${props => props.theme.spacingUnit * 4}px;
	font-size: ${props => props.theme.smallFontSize}px;

	.footer-container {
		display: flex;
		justify-content: flex-end;
	}
`;

export default StyledFooter;
