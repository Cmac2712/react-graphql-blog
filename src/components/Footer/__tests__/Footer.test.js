import React from 'react';
import { mount } from 'enzyme';
import wait from 'waait';
import toJSON from 'enzyme-to-json';
import Footer from '../index';

describe('<Footer/>', () => {

	it('should match snapshot', () => {
		const wrapper = mount(<Footer/>);

		expect(toJSON(wrapper)).toMatchSnapshot();
	});
	
	it('should say my name', () => {
		const wrapper = mount(<Footer/>);
		const myName = wrapper.find('.myname').text(); 

		expect(myName).toBe('Craig MacIntyre');
	});
	
	it('should say the correct year', () => {
		const wrapper = mount(<Footer/>);
		const yearInFooter = wrapper.find('.year').text(); 
		const currentYear = new Date().getFullYear().toString();

		expect(yearInFooter).toBe(currentYear);
	});

	it('should link to project on GitHub', () => {
		const wrapper = mount(<Footer/>);
		const projectGitHubUrl = 'https://github.com/Cmac2712/react-graphql-blog';
		const urlInFooter = wrapper.find('.github').prop('href');

		expect(urlInFooter).toBe(urlInFooter);
	});	
	
});
