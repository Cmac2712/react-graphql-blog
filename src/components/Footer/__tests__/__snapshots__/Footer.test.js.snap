// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should match snapshot 1`] = `
<Footer>
  <styled.footer>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": false,
            "lastClassName": "eJJGiC",
            "rules": Array [
              "
	padding: ",
              [Function],
              "px;
	background-color: #f1f1f1;
	margin-top: ",
              [Function],
              "px;
	font-size: ",
              [Function],
              "px;

	.footer-container {
		display: flex;
		justify-content: flex-end;
		align-items: center;
	}

	.github {
		display: flex;
		align-items: center;
		margin-right: auto;
	}

	.github-icon-container {
		font-size: 0;
		height: 18px;
		width: 18px;
	}

	.github span:first-child {
		margin-right: ",
              [Function],
              "px;
	}

	.myname {
		margin-right: ",
              [Function],
              "px;
	}

",
            ],
          },
          "displayName": "styled.footer",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "footer",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <footer
        className="sc-ifAKCX eJJGiC"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bdVaJa",
                  "isStatic": false,
                  "lastClassName": "hDbywE",
                  "rules": Array [
                    "
	margin: 0 auto;
	max-width: ",
                    [Function],
                    ";
	padding: 0 ",
                    [Function],
                    "px;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bdVaJa",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bdVaJa hDbywE"
            >
              <div
                className="footer-container"
              >
                <a
                  className="github"
                  href="https://github.com/Cmac2712/react-graphql-blog"
                >
                  <span>
                    View on GitHub
                  </span>
                  <ReactSVG
                    className="github-icon-container"
                    evalScripts="never"
                    fallback={null}
                    loading={null}
                    onInjected={[Function]}
                    renumerateIRIElements={true}
                    src="./images/github.svg"
                    svgClassName="github-icon"
                    svgStyle={
                      Object {
                        "fill": "red",
                        "height": 16,
                        "width": 16,
                      }
                    }
                    wrapper="span"
                  >
                    <span
                      className="github-icon-container"
                    />
                  </ReactSVG>
                </a>
                <a
                  className="myname"
                  href="https://craigmacintyre.co.uk/"
                >
                  Craig MacIntyre
                </a>
                <div
                  className="year"
                >
                  2019
                </div>
              </div>
            </div>
          </StyledComponent>
        </styled.div>
      </footer>
    </StyledComponent>
  </styled.footer>
</Footer>
`;
